rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,161 @@
+4. Write the segmented data into the log file of the corresponding column family.  
+
+#### Common API interface
+After the table columns are clustered, the writing process includes the process of sorting and splitting the data compared to the original bucket bucketing. A new append interface needs to be introduced to support column families.  
+Introduce ColumnFamilyAppendHandle extend AppendHandle to implement column family writing.
+
+![append-handle](append-handle.png)
+
+### Reading data
+#### ColumnFamilyReader and RowReader
+![row-reader](row-reader.png)
+
+Hudi internal row reader reading steps:  
+1. Hudi organizes files by column families to be read.
+2. Introduce familyReader to merge and read each column family's own baseFile and logfile to achieve column family-level data reading.  
+    * Since log files are written after being sorted by primary key, familyReader merges its own baseFile and logFile by primary key using sortMerge.
+    * familyReader supports upstream incoming column pruning to reduce IO overhead when reading data.  
+    * During the merging process, if the user specifies the precombie field for the column family, the merging strategy will be selected based on the precombie field. This logic reuses Hudi's own precombine logic and does not need to be modified.    
+3. Row reader merges the data read by multiple familyReaders according to the primary key.  
+
+Since the data read by each familyReader is sorted by the primary key, the row reader merges the data read by each familyReader in the form of sortMergeJoin and returns the complete data.  
+
+The entire reading process involves a large amount of data merging, but because the data itself is sorted, the memory consumption of the entire merging process is very low and the merging is fast. Compared with Hudi's native merging method, the memory pressure and the merging time are significantly reduced.
+
+#### Engine reads pseudo process
+![process-read](process-read.png)
+
+1) The engine itself delivers the data files that need to be scanned to executor/woker/taskmanger.  
+2) executor/worker/taskmanger calls Hudi’s rowReader interface and passes in column clipping and filter conditions to rowReader.  
+3) The Hudi kernel completes the data reading of rowReader and returns complete data. The data format is Avro.  
+4) The engine gets the Avro format data and needs to convert it into the data format it needs. For example, spark needs to be converted into unsaferow, hetu into block, flink into row, and hive into arrayWritable.
+
+### Column family level compaction
+Extend Hudi's compaction schedule module to merge each column family's own base file and log file:
+
+![family-compaction](family-compaction.png)
+
+### Full compaction
+Extend Hudi's compaction schedule module to merge and update all column families in the entire table.  
+After merging at the column family level, multiple column families are finally merged into a complete row and saved.  
```

Comment:
So after the full compaction, we can not prune columns based on family name for these files? Or you are saying we just utilize the parquet native column pruning.

Reply:
Yes, we just utilize the parquet native column pruning.

Reply:
The full compaction should be an optional use case in production I think, let's make it clear on the document.

Reply:
I agree, it is already written in line 164